File: Miscellaneous/stack.py
class Stack():
    def __init__(self, items=[]):
        self.items = items
    def is_empty(self):
        return self.items == []
    def push(self, obj):
        self.items.append(obj)
    def pop(self):
        rets = self.items[-1]
        self.items = self.items[:-1]
        return rets
    def peek(self):
        return self.items[-1]
    def size(self):
        return len(self.items)
    def __str__(self):
        rets = "["
        for i,v in enumerate(self.items):
            if i == self.size()-1:
                rets += str(v) +"]"
            else:
                rets += str(v) +","
        return rets

File: DsinJs/stack.js
class Stack
{
    constructor()
    {
        this.items = [];
    }

    push(element)
    {
        this.items.push(element);
    }
    pop()
    {
        return this.items.pop();
    }
    peek()
    {
        return this.items[this.items.length - 1];
    }
    isEmpty()
    {
        return this.items.length == 0;
    }
    size()
    {
        return this.items.length;
    }
    clear()
    {
        this.items = [];
    }
}

const stack = new Stack();
stack.push(12);
stack.push(15);
console.log(stack);
console.log(stack.peek()); // 15
stack.push(20);
console.log(stack.size());
console.log(stack.isEmpty());
stack.pop();
stack.pop();
console.log(stack.size()); //1
console.log(stack.peek()); // 12

